File: TOKI-Learning/c++/perkalian_matriks.cpp
#include <cstdio>

int main()
{
    int n, m, p, i, j, k, hasil;
    scanf("%d %d %d", &n, &m, &p);
    int matriks1[n][m], matriks2[m][p];
    for (i = 0; i < n; i++)
        for (j = 0; j < m; j++)
            scanf("%d", &matriks1[i][j]);
    for (i = 0; i < m; i++)
        for (j = 0; j < p; j++)
            scanf("%d", &matriks2[i][j]);
    for (i = 0; i < n; i++)
    {
        for (j = 0; j < p; j++)
        {
            hasil = 0;
            for (k = 0; k < m; k++)
            {
                hasil += matriks1[i][k] * matriks2[k][j];
            }
            if (j == p - 1)
                printf("%d\n", hasil);
            else
                printf("%d ", hasil);
        }
    }
    return 0;
}

File: TOKI-Learning/pascal/perkalian_matriks.pas
var
	matriks1, matriks2 : array [1..100,1..100] of byte;
	n, m, p, i, j, k : byte;
	tmp : longint;
	
begin
	readln(n,m,p);
	for i := 1 to n do
	begin
		for j := 1 to m do
			read(matriks1[i][j]);
		readln;
	end;
	
	for i := 1 to m do
	begin
		for j := 1 to p do
			read(matriks2[i][j]);
		readln;
	end;
	
	for i := 1 to n do
		for j := 1 to p do
		begin
			tmp := 0;
			for k := 1 to m do
				tmp := tmp + (matriks1[i][k]*matriks2[k][j]);
			write(tmp);
			if j < p then
				write(' ')
			else
				writeln;
		end;
end.